config/src/main/java/io/scalecube/config/ConfigRegistryImpl.java

```diff
@@ -598,7 +599,10 @@ public String valueOrThrow() {
         } catch (NoSuchFieldException e) {
           throw ThrowableUtil.propagate(e);
         }
-        fields.add(new ObjectPropertyField(field, bindingMap.get(fieldName)));
+        int modifiers = field.getModifiers();
+        if (!Modifier.isStatic(modifiers) && !Modifier.isFinal(modifiers)) {
```

Comment:
Why final? Maybe it make sense to left `final` so you can make config class immutable and still it will be initialized. Also, one thing to check is that we basically should handle those classes as they are immutable in the sense that - if field modified we create new instance of this class instead of modifying existing instance.

Reply:
> if field modified we create new instance of this class instead of modifying existing instance.

This is how it works: client creates `ObjectConfigProperty<MyConfig> configProperty = configRegistry.objectProperty(MyConfig.class)` , at this point the framework under the hood prepares `valueParser` and `propertyCallback` associated with `configProperty` instance,  then every time client calls  `configProperty.value()`  the framework creates new instance of object `MyConfig`.